src/main/java/frc/robot/subsystems/arm/ArmIOReal.java

```diff
@@ -0,0 +1,324 @@
+        shoulderMainMotor.setInverted(MAIN_CLOCKWISE);
+
+        elbowMainMotorConfig.Voltage.PeakForwardVoltage = ArmConstants.VOLT_COMP_SATURATION;
+        elbowMainMotorConfig.Voltage.PeakReverseVoltage = -ArmConstants.VOLT_COMP_SATURATION;
+        elbowMainMotorConfig.Slot1 = ELBOW_PID;
+        elbowMainMotorConfig.CurrentLimits.SupplyCurrentLimit = ArmConstants.CURRENT_LIMIT;
+        elbowMainMotorConfig.CurrentLimits.StatorCurrentLimit = ArmConstants.CURRENT_LIMIT;
+        elbowMainMotorConfig.Feedback.SensorToMechanismRatio = SENSOR_TO_MECHANISM_RATIO;
+        elbowMainMotorConfig.Feedback.RotorToSensorRatio = ROTOR_TO_SENSOR_RATIO;
+        elbowMainMotor.getConfigurator().apply(elbowMainMotorConfig);
+
+        elbowMainMotor.setNeutralMode(NeutralModeValue.Brake);
+        elbowMainMotor.setInverted(MAIN_CLOCKWISE);
+
+    }
+
+    /**
+     * Get the instance of the arm subsystem.
+     *
+     * @return Arm instance
+     */
+    public static ArmIOReal getInstance() {
+        if (INSTANCE == null) {
+            INSTANCE = new ArmIOReal();
+        }
+        return INSTANCE;
+    }
+
+
+    /**
+     * Sets the power of the shoulder motors.
+     *
+     * @param power desired power. [-1,1]
+     */
+    public void setShoulderJointPower(double power) {
+        shoulderMainMotor.setControl(new DutyCycleOut(power));
+    }
+
+    public double getShoulderJointPower() {
+        return shoulderMainMotor.get();
+    }
+
+    public double getElbowJointPower() {
+        return elbowMainMotor.get();
+    }
+
+    /**
+     * Sets the power of the elbow motors.
+     *
+     * @param power desired power. [-1,1]
+     */
+    public void setElbowJointPower(double power) {
+        elbowMainMotor.setControl(new DutyCycleOut(power));
+    }
+
+
+    /**
+     * Sets the angle of the shoulder joint.
+     *
+     * @param angle desired angle. [degrees]
+     */
+    public void setShoulderJointAngle(Rotation2d angle) {
+        setShoulderJointAngle(angle.getRadians(), SHOULDER_FEED_FORWARD_MULTIPLIER, SHOULDER_VELOCITY);
+    }
+
+
+    private void setShoulderJointAngle(double angle, double ffMultiplier, int velocity) {
+        angle = AngleUtil.normalize(angle);
+        Rotation2d error = new Rotation2d(angle).minus(new Rotation2d(currentShoulderAngle));
+        if (shoulderEncoder.isConnected()) {
+            motorControlRequest = new PositionVoltage(angle);
+            shoulderMainMotor.setControl(new PositionDutyCycle(angle, velocity, true,
+                    shoulderFeedforward * ffMultiplier, 0, true)
+                    .withEnableFOC(true));
+        } else {
+            shoulderMainMotor.stopMotor();
+        }
+    }
+
+
+    public Rotation2d getShoulderJointAngle() {
+        return Rotation2d.fromRotations(shoulderEncoder.getAbsolutePosition() - shoulderOffset);
+    }
+
+    public Rotation2d getElbowJointAngle() {
+        return Rotation2d.fromRotations(elbowEncoder.getAbsolutePosition() - elbowOffset);
```

Comment:
It's better to set the offset and ration once instead of calculating it yourself. You can find the relevant methods [here](https://github.wpilib.org/allwpilib/docs/release/java/edu/wpi/first/wpilibj/DutyCycleEncoder.html).